Simplify tests using example terraform and pull in boilerplate pieces

dynamike opened the pull request:
This adopts some of the new patterns around how we are using terratest. It includes much of the boilerplate bits (e.g., Makefile) and breaks the tests into 2 parts.
1) For simple testing of the module
2) Adding a custom bucket policy

kilbergr commented:
I have been running this again, and finding the following failure:

```Error: Error creating S3 bucket: IllegalLocationConstraintException: The ap-northeast-1 location constraint is incompatible for the region specific endpoint this request was sent to.``` 
(with different regions of course). I wonder whether I just got lucky the first time I ran.

dynamike commented:
@kilbergr I think the errors you were seeing were from previously failed terraform runs. I've moved the changed to logic to generate state files in temporary paths and also test it manually against ap-northeast-1 . 

I also updated the docs.
